build.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python
+
+import argparse
+import contextlib
+import datetime
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+import textwrap
+
+
+@contextlib.contextmanager
+def mkdtemp(*args, **kwargs):
+    tmp_dir = tempfile.mkdtemp(*args, **kwargs)
+    yield tmp_dir
+    shutil.rmtree(tmp_dir)
+
+
+def rmmkdir(dn):
+    if os.path.exists(dn):
+        shutil.rmtree(dn)
+    os.makedirs(dn)
+
+
+def git_commit_time(ref="HEAD"):
+    git_time_str = subprocess.check_output(
+        ["git", "show", "-s", "--format=%ci", ref],
+        universal_newlines=True
+    )
+    git_time_str = git_time_str.strip()
+    git_utctime = datetime.datetime.strptime(
+        git_time_str[:19] + " UTC",
+        "%Y-%m-%d %H:%M:%S %Z"
+    )
+
+    git_timedelta = datetime.timedelta()
+    if len(git_time_str) > 19:
+        git_timedelta = datetime.datetime.strptime(git_time_str[21:], "%H%M")
+        git_timedelta = datetime.timedelta(
+            hours=git_timedelta.hour, minutes=git_timedelta.minute
+        )
+        if git_time_str[20] == "-":
+            git_timedelta = -git_timedelta
+
+    git_time = git_utctime - git_timedelta
```

Comment:
Adjust the UTC time based on the timezone as explained above. Need to subtract as a negative delta means we are behind UTC and a positive delta means we are ahead of UTC.